Backend/app/rider-platform/rider-app/Main/src/API/Beckn/FRFS/OnConfirm.hs

```diff
@@ -55,22 +50,14 @@ onConfirm _ req = withFlowHandlerAPI $ do
   logDebug $ "Received OnConfirm request" <> encodeToText req
   withTransactionIdLogTag' transaction_id $ do
     dOnConfirmReq <- ACL.buildOnConfirmReq req
-    if isJust dOnConfirmReq
-      then do
-        let onConfirmReq = fromJust dOnConfirmReq
+    case dOnConfirmReq of
+      Just onConfirmReq -> do
         (merchant, booking) <- DOnConfirm.validateRequest onConfirmReq
         Metrics.finishMetrics Metrics.CONFIRM_FRFS merchant.name transaction_id booking.merchantOperatingCityId.getId
         fork "onConfirm request processing" $
           Redis.whenWithLockRedis (onConfirmProcessingLockKey onConfirmReq.bppOrderId) 60 $
             DOnConfirm.onConfirm merchant booking onConfirmReq
-      else do
-        frfsConfig <-
-          CQFRFSConfig.findByMerchantOperatingCityId ticketBooking.merchantOperatingCityId
-            >>= fromMaybeM (InternalError $ "FRFS config not found for merchant operating city Id " <> show ticketBooking.merchantOperatingCityId)
-        void $ QFRFSTicketBooking.updateStatusById DFRFSTicketBooking.FAILED (Id bookingId)
-        void $ QFRFSTicketBookingPayment.updateStatusByTicketBookingId DFRFSTicketBookingPayment.REFUND_PENDING (Id bookingId)
-        void $ DACFOC.callBPPCancel ticketBooking bapConfig frfsConfig Spec.CONFIRM_CANCEL ticketBooking.merchantId
-
+      Nothing -> DOnConfirm.onConfirmFailure bapConfig ticketBooking
```

Comment:
What if cancellation is not supported for Bus Booking?

Reply:
Throwing CancellationNotSupported Error in the APIs that can cancel the Booking.
```
endpoint: /frfs/booking/{bookingId}/canCancel
endpoint: /frfs/booking/{bookingId}/cancel
```